src/components/api/json-rpc/che-json-rpc-master-api.ts

```diff
@@ -126,8 +134,13 @@ export class CheJsonRpcMasterApi {
   }
 
   reconnect(entrypoint: string): void {
-    this.$log.warn('WebSocket connection is closed.');
-    if (this.reconnectionAttemptNumber === this.maxReconnectionAttempts) {
+    if (this.reconnectionAttemptNumber < this.maxReconnectionAttempts) {
+      this.$log.warn('WebSocket connection is closed.');
+    } else {
+      this.globalWarningBannerService.addMessage(`WebSocket connections "${entrypoint}" are failing due to network restrictions.
```

Comment:
@RickJWagner @l0rd Would you like to validate that error message? The result is available in the PR description as screenshot.